wp-theme-2018/sidebar.php

```diff
@@ -11,20 +11,74 @@
 
 global $EPFL_MENU_LOCATION;
 
+function get_rendered_sider_item($crumb_item, $currentPage, $children) {
+    $title = $crumb_item['title'] ?? '';
+    $url = $crumb_item['url'] ?? '';
 
-// to display correctly the menu on level 1 pages, on missing menu, or for the blog home,
-// we need to add '.current-menu-parent' to the wrapper
-$classes = '';
+    if ($crumb_item['title'] == $currentPage->title) {
+        $subMenus = '<ul class=\"sub-menu\">';
+        foreach ($children as $child) {
+            $subMenus = $subMenus . "<li class=\"menu-item\"><a href=\"{$child['url']}\">{$child['title']}</a></li>";
+        }
+        $subMenus = $subMenus . '</ul>';
+        return "
+            <li class=\"menu-item active\">
+                <a href=\"{$url}\" title=\"{$title}\">
+                    {$title}
+                </a>
+                {$subMenus}
+            </li>";
+    } else {
+        return "
+            <li class=\"menu-item\">
+                <a href=\"{$url}\" title=\"{$title}\">
+                    {$title}
+                </a>
+            </li>";
+    }
+}
+
+function getStitchedMenus($homePageUrl, $urlSite, $lang): array
+{
+    $main_post_page = get_option('page_for_posts');
+    if (! function_exists("pll_get_post")) {
+        # Menus and siblings require Polylang.
+        return [];
+    }
+    $current_language_page_id = pll_get_post($main_post_page, $lang);
+    $mainPostPageName = urlencode(get_the_title($current_language_page_id));
+    $mainPostPageUrl = get_permalink($current_language_page_id);
+
+    $urlApi = 'http://' . getenv('MENU_API_HOST') . ':3001/menus/getStitchedMenus/?lang=' . $lang . '&url=' . trailingslashit( $urlSite ) .
+        '&pageType=' . get_post_type() .
+        ($main_post_page == 0 ? '' : ($mainPostPageName == '' ? '' : '&mainPostPageName=' . $mainPostPageName)) .
+        ($main_post_page == 0 ? '' : ($mainPostPageUrl == '' ? '' : '&mainPostPageUrl=' . $mainPostPageUrl)).
+        '&postName=' . urlencode(get_the_title()) .
+        '&homePageUrl=' . $homePageUrl;
+    $curl = curl_init($urlApi);
+    curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
 
-$items = wp_get_nav_menu_items(get_current_menu_slug());
-$current_menu_entry = get_menu_entry_from_element_id($items, get_currently_viewed_element_id());
+    $response = curl_exec($curl);
+    if (curl_errno($curl)) {
```

Comment:
There's too many cases for this error management code to reason about. I'll push a fix.